packages/realm/src/app-services/SyncConfiguration.ts

```diff
@@ -146,20 +148,16 @@ export type SyncConfiguration = FlexibleSyncConfiguration | PartitionSyncConfigu
 
 /** @internal */
 export function toBindingSyncConfig(config: SyncConfiguration): binding.SyncConfig_Relaxed {
-  if (config.flexible) {
-    throw new Error("Flexible sync has not been implemented yet");
-  }
-  const { user, onError, _sessionStopPolicy, customHttpHeaders, clientReset } = config;
-  assert.instanceOf(user, User, "user");
-  validatePartitionValue(config.partitionValue);
```

Comment:
The above assertions have been removed.  Are they covered elsewhere?  I would assume `validateSyncConfiguration` should be called.

Reply:
Yes the validation is moved to `Configuration.ts` and `SyncConfiguration.ts`, and the calling of those methods are done in user-facing methods that handle configuration objects. Thus, `toBindingSyncConfig` assumes a validated config.

I do see that e.g. our exported methods on the `Sync` namespace in `Sync.ts` do not validate the arguments. The `getSyncSession()` method there are calling `toBindingSyncConfig()` so I'll add a call to `validateSyncConfig()` there 👍

Reply:
Great, thanks for the explanation 🙂

Reply:
No problem! 🙂